src/main/java/it/sweven/blockcovid/services/UserService.java

```diff
@@ -20,11 +22,14 @@ public User save(User user) {
     return userRepository.save(user);
   }
 
-  public Optional<User> getByToken(Token token) {
-    return userRepository.findByToken(token);
-  }
-
   public Optional<User> getByUsername(String username) {
```

Comment:
we should keep the `Optional` semantics in the repositories only, and implement services such that they throw errors if no data can be fetched (`.orElseThrow(() -> ...)`) to keep things consistent